edit/sections-editor.js

```diff
@@ -80,6 +81,45 @@ function createSectionsEditor({style, onTitleChanged}) {
     getSearchableInputs,
   };
 
+  function fitToContent(section) {
+    if (section.cm.isRefreshed) {
+      resize();
+    } else {
+      section.cm.on('update', resize);
+    }
+
+    function resize() {
+      let contentHeight = section.el.querySelector('.CodeMirror-sizer').offsetHeight;
+      if (contentHeight < section.cm.defaultTextHeight()) {
+        return;
+      }
+      contentHeight += 9; // border & resize grip
+      section.cm.off('update', resize);
+      const cmHeight = section.cm.getWrapperElement().offsetHeight;
+      const maxHeight = cmHeight + window.innerHeight - section.el.offsetHeight;
+      section.cm.setSize(null, Math.min(contentHeight, maxHeight));
+      if (sections.every(s => s.cm.isRefreshed)) {
+        fitToAvailableSpace();
+      }
+      setTimeout(() => {
+        container.classList.add('section-editor-ready');
```

Comment:
When there are multiple editors, it is tricky to decide when to show (unhide) them:

* It is possible that when the first editor is resized, other editors are not resized yet.
* We can't show all editors immediately when one editor is resized or we will get "expanding animation" for other editors.
* We can't show the current editor immediately or it may be pushed down when the previous editor is shown.
* We can't wait until all editors are resized since editors outside of the window are not rendered.

The current logic is to show all editors after 50ms of the first editor is resized. The correct way to handle this problem is to grab a list of sections inside the current view and wait until they are resized. However, I'm not sure if there is an accurate method to mimic `IntersectionObserver` as we used in refresh-on-view.js.

Reply:
IDK man, but I can tell you that the delay we're looking to create to avoid the brief flash of editors expanding is so tiny, that losing the `setTimeout` wrapper and just revealing on the added class eliminates it completely, without any help. Checked the bulkiest styles, and also made a test style with a bunch of short sections.

Reply:
I remember at one point someone requested that we put the separate sections into tabs to make it easier to navigate. What do you think about implementing that now?

Reply:
Do you mean #68? But how does it relate to this PR?

Reply:
Having tabs would mean you wouldn't need to worry about resizing the editor height. I mean, each block would fill the height of the window, then we can remove the height resizing code.

Reply:
On the other hand, there is this case where there are so many applies to entries that the editor ends up relatively small. We'd have to limit the height or move the applies to block into a collapsible panel to make it work for a fixed height tab layout.

![2019-06-18 09_05_35-Google - Clean Dark](https://user-images.githubusercontent.com/136959/59691633-ec823400-91a8-11e9-86e4-e141e50946d1.png)

Reply:
Here are some pros and cons of force-setting the editor height using CSS: https://github.com/openstyles/stylus/pull/722#issuecomment-498027364

If we want to add a tab layout, it should be implemented in another PR. This PR is intended to fix #722.